### PR TITLE
Blowing kisses at the supermatter crystal will now power it.

### PR DESCRIPTION
**Pull request by Pickle-Coding**
## About The Pull Request
Kisses blown via the *kiss emote will power the supermatter crystal.

## Why It's Good For The Game
Adds a nice way to start the engine if emitters don't fire.

## Changelog

:cl:
add: Blown kisses will now power the supermatter engine.
/:cl:

### CODE REVIEW

**Comment by Colovorat** [comment]
Please say you tested it intensively

**Comment by Pickle-Coding** [comment]
I blew a kiss and it got powered. Also tried with kiss of death, though maybe I could make that one make the SM angry.

**Comment by nianjiilical** [comment]
only if the sm becomes flushed if it gets hit by a kiss while not turned on

**Comment by Pickle-Coding** [comment]
It now sets psyCoeff to 1 and set the overlay so it becomes flushed. Kiss of death will give it a lot of power.

**Comment by wesoda25** [comment]
The power of love...

File: code/modules/power/supermatter/supermatter.dm [hunk 1]
```diff
@@ -852,13 +852,17 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 
 /obj/machinery/power/supermatter_crystal/bullet_act(obj/projectile/projectile)
 	var/turf/local_turf = loc
+	var/kiss_power = istype(projectile, /obj/projectile/kiss/death) ? 20000 : istype(projectile, /obj/projectile/kiss) ? 100 : 0
```
Comment (JohnFulpWillard) [inline code/modules/power/supermatter/supermatter.dm:855]: Why is the second ? needed

Comment (Pickle-Coding) [reply code/modules/power/supermatter/supermatter.dm:855]: So different kisses give SM different power.

**Comment by Farquaar** [comment]
I dunno. I feel like this should be a superpower, or only a wizard/Valentines day thing.

File: code/modules/power/supermatter/supermatter.dm (hunk 1 above)
Comment (Jcarr0ndo) [reply code/modules/power/supermatter/supermatter.dm:855]: > So different kisses give SM different power.

What if you make it start gracefully if blown a kiss by an Engineer or the CE (more power). But if anyone else sends a kiss it also starts but at 99% integrity?

**Comment by Coastfront** [comment]
Okay but when can we then meet the supermatter at the office, Take them out on a few dinner dates, and then do cock and ball torture sessions every 10pm 24/7?

File: code/modules/power/supermatter/supermatter.dm (hunk 1 above)
Comment (Mothblocks) [inline code/modules/power/supermatter/supermatter.dm:855]: Make this a switch instead this is very hard to read

**Comment by optimumtact** [comment]
how much does a regular emitter shot give?

**Comment by Colovorat** [comment]
> how much does a regular emitter shot give?

Uh oh, fun police

**Comment by Pickle-Coding** [comment]
> how much does a regular emitter shot give?

About 60 I think. The value I chose for the kiss is pretty arbitrary so changing it won't be much of a problem if needed.

**Comment by optimumtact** [comment]
it should be about the same as an emitter shot

**Comment by iprice** [comment]
This is awesome, love the idea.  Also gives a nice alternative to throwing engineering "stuff" into the engine when you have to jump start it.

**Pull request merged**